frontend/src/libs/components/protected-route/protected-route.tsx

```diff
@@ -0,0 +1,32 @@
+import { Navigate } from 'react-router-dom';
+import { type UserGroupKey } from 'shared/build/index.js';
+
+import { AppRoute, DataStatus } from '~/libs/enums/enums.js';
+import { useAppSelector } from '~/libs/hooks/hooks.js';
+import { type ValueOf } from '~/libs/types/types.js';
+import { selectIsLoading, selectUser } from '~/slices/auth/selectors.js';
+
+import { RouterOutlet, Spinner } from '../components.js';
```

Comment:
That's circular dependency. You can re-write it as:
from '../spinner/spinner.js'
from '../router/router.js' // yeah, let's import RouterOutlet and Navigate there.

Reply:
Did I understand correctly that if a component is exported from components.ts, then inside this component I should not import from components.ts?

Reply:
Yes, that's correct 👍